admin-dev/themes/new-theme/js/components/modal/progress-modal.ts

```diff
@@ -0,0 +1,421 @@
+    this.dialog.classList.add('modal-dialog');
+    if (params.dialogStyle) {
+      Object.keys(params.dialogStyle).forEach((key: string) => {
+        // @ts-ignore
+        this.dialog.style[key] = params.dialogStyle[key];
+      });
+    }
+
+    this.container.appendChild(this.dialog);
+  }
+}
+
+export class ProgressModalContainer {
+  footer!: HTMLElement;
+
+  container!: HTMLElement;
+
+  dialog!: HTMLElement;
+
+  content!: HTMLElement;
+
+  message!: HTMLElement;
+
+  header!: HTMLElement;
+
+  title?: HTMLElement;
+
+  body!: HTMLElement;
+
+  stopProcessingButton!: HTMLElement;
+
+  switchToErrorButton!: HTMLElement;
+
+  progressDone!: HTMLElement;
+
+  progressPercentage!: HTMLElement;
+
+  errorCount!: HTMLElement;
+
+  constructor(params: ProgressModalParams) {
+    this.buildModalContainer(params);
+  }
+
+  buildModalContainer(params: ProgressModalParams): void {
+    // Modal content element
+    this.content = document.createElement('div');
+    this.content.classList.add('modal-content');
+
+    // Modal message element
+    this.message = document.createElement('p');
+    this.message.classList.add('modal-message');
+
+    // Modal header element
+    this.header = document.createElement('div');
+    this.header.classList.add('modal-header');
+
+    params.modalTitle = params.modalTitle.replace('%d', params.total.toString());
+
+    this.progressPercentage = document.createElement('div');
+    this.progressPercentage.classList.add(
+      'float-right',
+      'progress-percent',
+    );
+    this.progressPercentage.append('0%');
+
+    this.body = document.createElement('div');
+    this.body.classList.add('modal-body', 'text-left', 'font-weight-normal');
+
+    this.body.append(this.progressPercentage);
+    this.body.append(this.buildProgressBar());
+
+    this.switchToErrorButton = document.createElement('div');
+    this.switchToErrorButton.classList.add(
+      'switch-to-errors-button',
+      'alert',
+      'alert-warning'
+    );
+
+    this.errorCount = document.createElement('span');
+    this.errorCount.classList.add(
+      'modal-error-count'
+    )
+    this.errorCount.innerHTML = '0';
+
+    let errorLogString = 'View %d error logs';
+    this.switchToErrorButton.innerHTML = errorLogString.replace('%d', this.errorCount.outerHTML);
+
+    this.header.append(this.switchToErrorButton);
+
+    let errorContainer = document.createElement('div');
+    errorContainer.classList.add('modal-error-container-single');
+    this.body.append(errorContainer);
+    // Modal footer element
+    this.footer = document.createElement('div');
+    this.footer.classList.add('modal-footer');
+
+    this.stopProcessingButton = document.createElement('button');
+    this.stopProcessingButton.setAttribute('type', 'button');
+    this.stopProcessingButton.classList.add('btn', 'btn-outline-secondary', 'btn-lg', 'stop-processing');
+    this.stopProcessingButton.innerHTML = 'Stop processing';
```

Comment:
We'll handle the multilang labels in another PR, so hard-coded values are ok for now